certificateutil/certificateutil.go

```diff
@@ -0,0 +1,208 @@
+			pems = append(pems, fmt.Sprintf("Bag Attributes\n%s", pem))
+		}
+	}
+
+	for _, pem := range pems {
+		cmd := command.New("openssl", "x509", "-noout", "-enddate", "-subject")
+		cmd.SetStdin(bytes.NewReader([]byte(pem)))
+		out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+		if err != nil {
+			return []CertificateInfosModel{}, fmt.Errorf("failed to read certificate infos, out: %s, error: %s", out, err)
+		}
+
+		lines := strings.Split(out, "\n")
+		if len(lines) < 2 {
+			return []CertificateInfosModel{}, fmt.Errorf("failed to parse certificate infos")
+		}
+
+		certificateInfos := CertificateInfosModel{}
+
+		// notAfter=Aug 15 14:15:19 2018 GMT
+		endDateLine := strings.TrimSpace(lines[0])
+		certificateInfos.RawEndDate = endDateLine
+		endDatePattern := `notAfter=(?P<date>.*)`
+		endDateRe := regexp.MustCompile(endDatePattern)
+		if matches := endDateRe.FindStringSubmatch(endDateLine); len(matches) == 2 {
+			endDateStr := matches[1]
+			endDate, err := time.Parse("Jan 2 15:04:05 2006 MST", endDateStr)
+			if err == nil {
+				certificateInfos.EndDate = endDate
+			} else {
+				log.Warnf("Failed to parse certificate endDate, error: %s", err)
+			}
+		} else {
+			log.Warnf("Failed to find pattern in %s, matches: %d", endDateLine, len(matches))
+		}
+
+		// subject= /UID=5KN/CN=iPhone Developer: Bitrise Bot (T36)/OU=339/O=Bitrise Bot/C=US
+		subjectLine := strings.TrimSpace(lines[1])
+		certificateInfos.RawSubject = subjectLine
+		certificateInfos.IsDevelopement = (strings.Contains(subjectLine, "Developer:") || strings.Contains(subjectLine, "Development:"))
+		subjectPattern := `subject= /UID=(?P<userID>.*)/CN=(?P<commonName>.*)/OU=(?P<teamID>.*)/O=(?P<name>.*)/C=(?P<local>.*)`
+		subjectRe := regexp.MustCompile(subjectPattern)
+		if matches := subjectRe.FindStringSubmatch(subjectLine); len(matches) == 6 {
+			userID := matches[1]
+			commonName := matches[2]
+			teamID := matches[3]
+			name := matches[4]
+			local := matches[5]
+
+			certificateInfos.UserID = userID
+			certificateInfos.CommonName = commonName
+			certificateInfos.TeamID = teamID
+			certificateInfos.Name = name
+			certificateInfos.Local = local
+			certificateInfos.IsDevelopement = (strings.Contains(commonName, "Developer:") || strings.Contains(commonName, "Development:"))
+		} else {
+			log.Warnf("Failed to find pattern in %s, matches: %d", subjectLine, len(matches))
+		}
+
+		certInfoModels = append(certInfoModels, certificateInfos)
+	}
+
+	return certInfoModels, nil
+}
+
+// CertificateInfosFromDerContent ...
+func CertificateInfosFromDerContent(pemContent []byte) (CertificateInfosModel, error) {
+	cmd := command.New("openssl", "x509", "-noout", "-enddate", "-subject", "-inform", "DER")
+	cmd.SetStdin(bytes.NewReader(pemContent))
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return CertificateInfosModel{}, fmt.Errorf("failed to read certificate infos, out: %s, error: %s", out, err)
+	}
+
+	lines := strings.Split(out, "\n")
```

Comment:
this output processing is the same as in `CertificateInfosFromPemContent`, please create a common function for this.